corehq/apps/users/middleware.py

```diff
@@ -45,9 +51,14 @@ def process_view(self, request, view_func, view_args, view_kwargs):
         if request.user.is_anonymous and 'domain' in view_kwargs:
             if ANONYMOUS_WEB_APPS_USAGE.enabled(view_kwargs['domain']):
                 request.couch_user = CouchUser.get_anonymous_mobile_worker(request.domain)
+        if determine_authtype_from_header(request) in (BASIC, DIGEST, API_KEY) and 'domain' in view_kwargs:
```

Comment:
I am sure you have thought of it already but just confirming,
why is this does not include `request.user.is_authenticated` or `request.user.is_anonymous`. If the user is not authenticated this should not be needed anyway?

Reply:
Also are we targetting web requests by doing `(BASIC, DIGEST, API_KEY)`? if yes, then can just do `not in TOKEN`?

Reply:
The reason we need this PR is to check whether a user has permission to see data outside their location before they can be authenticated. For full context, check out [FB 262139](http://manage.dimagi.com/default.asp?262139).

Reply:
We started with `determine_authtype_from_header(request) in (BASIC, DIGEST)` and added `API_KEY` to resolve failing tests. I don't see why we can't just support anything that will give us an unauthenticated username. i.e. `determine_authtype_from_header(request, default='NONE') != 'NONE'`. I'll try that out.

Reply:
^^^ It turns out that will involve a little more work. Not a bad idea, but I feel out of scope of this fix.

Reply:
>The reason we need this PR is to check whether a user has permission to see data outside their location before they can be authenticated

Oh cool.. so in case i try to login as a user, say iamnotmk@dimagi.com, with probably the wrong password, this would check location access for iamnotmk@dimagi.com but eventually fail login? Or the use case is for logged in users to this case would not happen?